python/benchmark/gen_data_distributed.py

```diff
@@ -144,6 +142,98 @@ def make_blobs_udf(iter: Iterable[pd.DataFrame]) -> Iterable[pd.DataFrame]:
         )
 
 
+class LowRankMatrixDataGen(DataGenBase):
+    """Generate random dataset using a distributed version of sklearn.datasets.make_low_rank_matrix,
+    which creates large low rank matrices for benchmarking dimensionality reduction algos like pca
+    """
+
+    def __init__(self, argv: List[Any]) -> None:
+        super().__init__()
+        self._parse_arguments(argv)
+
+    def _supported_extra_params(self) -> Dict[str, Any]:
+        params = inspect_default_params_from_func(
+            make_low_rank_matrix, ["n_samples", "n_features"]
+        )
+        # must replace the None to the correct type
+        params["random_state"] = int
+        return params
+
+    def gen_dataframe(self, spark: SparkSession) -> Tuple[DataFrame, List[str]]:
+        dtype = self.dtype
+        params = self.extra_params
+
+        if "random_state" not in params:
+            # for reproducible dataset.
+            params["random_state"] = 1
+
+        print(f"Passing {params} to make_low_rank_matrix")
+
+        rows = self.num_rows
+        cols = self.num_cols
+        assert self.args is not None
+        num_partitions = self.args.output_num_files
+
+        # Set num_partitions to Spark's default if output_num_files is not provided.
+        if num_partitions is None:
+            num_partitions = spark.sparkContext.defaultParallelism
+
+        generator = check_random_state(params["random_state"])
+        n = min(rows, cols)
+        # If params not provided, set to defaults.
+        effective_rank = params.get("effective_rank", 10)
+        tail_strength = params.get("tail_strength", 0.5)
+
+        partition_sizes = [rows // num_partitions] * num_partitions
+        partition_sizes[-1] += rows % num_partitions
+        # Check sizes to ensure QR decomp produces a matrix of the correct dimension.
+        for size in partition_sizes:
```

Comment:
it seems checking the last two values is enough

Reply:
I think just checking any but the last (e.g. just the first) should be enough, as last is strictly >= non-last